File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/recipes/RecipeWithMaxInteractionCount.kt
package examples.kotlin.recipes

import com.ing.baker.recipe.kotlindsl.ExperimentalDsl
import com.ing.baker.recipe.kotlindsl.recipe
import examples.kotlin.interactions.ShipOrder

@ExperimentalDsl
object RecipeWithMaxInteractionCount {
    val recipe = recipe("example") {
        interaction<ShipOrder> {
            maximumInteractionCount = 1
        }
    }
}

File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/RecipeWithMaxInteractionCount.scala
package examples.scala.recipes

import com.ing.baker.recipe.scaladsl.Recipe
import examples.scala.interactions.ShipOrder

object RecipeWithMaxInteractionCount {
  val recipe: Recipe = Recipe("example")
    .withInteraction(
      ShipOrder.interaction
        .withMaximumInteractionCount(1)
    )
}